RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -233,57 +253,71 @@ private struct PresentingPaywallModifier: ViewModifier {
 
     @State
     private var data: Data?
-
+    
     func body(content: Content) -> some View {
-        content
-            .sheet(item: self.$data, onDismiss: self.onDismiss) { data in
-                PaywallView(
-                    configuration: .init(
-                        content: self.content,
-                        customerInfo: data.customerInfo,
-                        fonts: self.fontProvider,
-                        displayCloseButton: true,
-                        introEligibility: self.introEligibility,
-                        purchaseHandler: self.purchaseHandler
-                    )
-                )
-                .onPurchaseStarted {
-                    self.purchaseStarted?()
-                }
-                .onPurchaseCompleted {
-                    self.purchaseCompleted?($0)
-                }
-                .onPurchaseCancelled {
-                    self.purchaseCancelled?()
-                }
-                .onRestoreCompleted { customerInfo in
-                    self.restoreCompleted?(customerInfo)
-
-                    if !self.shouldDisplay(customerInfo) {
-                        self.close()
+        Group {
+            switch presentationMode {
+            case .sheet:
+                content
+                    .sheet(item: self.$data, onDismiss: self.onDismiss) { data in
+                        paywallView(data: data)
```

Comment:
Explicit `self` per our style guide:
```suggestion
                        self.paywallView(data: data)
```